integrationtests/src/test/java/org/axonframework/integrationtests/eventhandling/TrackingEventProcessorTest.java

```diff
@@ -1065,6 +1083,8 @@ void testDoubleSplitAndMerge() throws Exception {
 
         waitForProcessingStatus(0, EventTrackerStatus::isCaughtUp);
 
+        assertFalse(testSubject.processingStatus().get(0).isMerging());
+
         assertTrue(testSubject.mergeSegment(0).join(), "Expected merge to succeed");
```

Comment:
Why not check that after this mergeSegment call that the status `isMerging` returns true?

Reply:
due to a race condition is really hard to test this into this particular test case. Most of the time the test will fail.
We will think about a better strategy to avoid this race condition that occurs from time to time also in CI env.